packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2023,12 +2023,28 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     // as the pre-formatting value of the previous pass (repeat call).
     final bool textChanged = _value?.text != value?.text;
     final bool isRepeat = value == _lastFormattedUnmodifiedTextEditingValue;
+    final bool isRepeatText = value?.text == _lastFormattedUnmodifiedTextEditingValue?.text;
 
-    if (textChanged && widget.inputFormatters != null && widget.inputFormatters.isNotEmpty) {
+    // There's no need to format when the value start to composing and repeat composing.
+    final bool isValueComposing = value != null && value.composing != const TextRange(start: -1, end: -1);
```

Comment:
IIRC `TextRange.isValid` does the same:
```suggestion
    final bool isValueComposing = value?.composing?.isValid ?? false;
```